etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/orders/SendOrderUseCase.java

```diff
@@ -22,44 +22,44 @@ public static SendOrderUseCase getInstance() {
         return INSTANCE;
     }
 
-    public void convertAndSend(final Order<?> order, String submissionId)
+    public void convertAndSend(final Order<?> order, String receivedSubmissionId)
             throws UnableToSendOrderException {
 
-        savePartnerMetadataForReceivedOrder(submissionId, order);
+        savePartnerMetadataForReceivedOrder(receivedSubmissionId, order);
 
         var omlOrder = converter.convertMetadataToOmlOrder(order);
         metadata.put(order.getFhirResourceId(), EtorMetadataStep.ORDER_CONVERTED_TO_OML);
+
         omlOrder = converter.addContactSectionToPatientResource(omlOrder);
         metadata.put(order.getFhirResourceId(), EtorMetadataStep.CONTACT_SECTION_ADDED_TO_PATIENT);
-        sender.sendOrder(omlOrder);
 
-        saveSentOrderSubmissionId(
-                submissionId,
-                "TBD, need to be filled in from the sender.sendOrder(omlOrder) call",
-                order);
+        String sentSubmissionId = sender.sendOrder(omlOrder).orElse(null);
+
+        saveSentOrderSubmissionId(receivedSubmissionId, sentSubmissionId);
     }
 
-    private void savePartnerMetadataForReceivedOrder(String submissionId, final Order<?> order) {
-        if (submissionId == null) {
+    private void savePartnerMetadataForReceivedOrder(
+            String receivedSubmissionId, final Order<?> order) {
+        if (receivedSubmissionId == null) {
             return;
```

Comment:
Do we need to log in these null SubmissionId scenarios? Just so we have awareness?

Reply:
Yes, good point